Makefile

```diff
@@ -13,7 +13,7 @@ MINIMUM_SUPPORTED_GO_MAJOR_VERSION = 1
 MINIMUM_SUPPORTED_GO_MINOR_VERSION = 15
 GO_VERSION_VALIDATION_ERR_MSG = Golang version is not supported, please update to at least $(MINIMUM_SUPPORTED_GO_MAJOR_VERSION).$(MINIMUM_SUPPORTED_GO_MINOR_VERSION)
 
-DOCKER_BUILD := docker build --build-arg VERSION=${VERSION}
+DOCKER_BUILD := docker buildx build --platform linux/amd64,linux/arm64,linux/arm/v6 --build-arg VERSION=${VERSION}
```

Comment:
This causes a funny interaction with the building of images that are explicitly tagged per architecture (https://github.com/oauth2-proxy/oauth2-proxy/blob/633734f93d2255da9861ed16819f0daea53b0f41/Makefile#L48-L56).

Maybe those lines would be better off being removed?

Reply:
Initially, we will want to have the images tagged for both the normal tags (ie release number and latest) as well as the arch specific tags, to allow people to migrate across. Can we do that from the same build at all?

How does the go build within the dockerfile understand which architecture it should be building for? IIRC we set the GOOS and GOARCH specifically don't we?

Reply:
Indeed, I meant to ask about deletion of those other builds, but didn't want to lead with code removal.

We could retag latest tag to the matrix of supported tags if that sounds good?

I don't understand how buildkit makes it work, but I haven't had the need to update GOOS and GOARCH when building multiarch containers from go binaries using `docker buildx build`

Reply:
Perhaps in the initial link, `TARGETOS` and `TARGETARCH` are implicitly set to a set of build flags, `GOARCH` and `GOOS` being in that set

Reply:
It can be explicitly passed through into the `Dockerfile` like so if that's preferable:
https://www.docker.com/blog/containerize-your-go-developer-environment-part-1/

Reply:
What if instead, we set the build and push commands to use buildx, but then set the platform for each of the lines in the later parts of the dockerfile, eg
```
docker:
	$(DOCKER_BUILD) --platform linux/amd64,linux/arm64,linux/arm/v6 -f Dockerfile -t $(REGISTRY)/oauth2-proxy:latest .

docker-all: docker
	$(DOCKER_BUILD) --platfom linux/amd64 -f Dockerfile -t $(REGISTRY)/oauth2-proxy:latest-amd64 .
	$(DOCKER_BUILD) --platfom linux/arm64 -f Dockerfile -t $(REGISTRY)/oauth2-proxy:latest-arm64 .
```
Does this kind of thing work?

Reply:
Sorry for the long poll. That sounds good to me, I'll work on an update